folsom/src/test/java/com/spotify/folsom/MemcachedServer.java

```diff
@@ -38,34 +40,64 @@ public enum AuthenticationMode {
       Suppliers.memoize(MemcachedServer::new)::get;
 
   public static int DEFAULT_PORT = 11211;
+
+  private static final String MEMCACHED_VERSION = "1.6.21";
   private final String username;
   private final String password;
+  private final boolean useTLS;
 
   public MemcachedServer() {
     this(null, null);
   }
 
   public MemcachedServer(String username, String password) {
-    this(username, password, DEFAULT_PORT, AuthenticationMode.NONE);
+    this(username, password, DEFAULT_PORT, AuthenticationMode.NONE, false);
   }
 
   public MemcachedServer(String username, String password, AuthenticationMode authenticationMode) {
-    this(username, password, DEFAULT_PORT, authenticationMode);
+    this(username, password, DEFAULT_PORT, authenticationMode, false);
+  }
+
+  public MemcachedServer(boolean useTLS) {
+    this(null, null, DEFAULT_PORT, AuthenticationMode.NONE, useTLS);
   }
 
   public MemcachedServer(
-      String username, String password, int port, AuthenticationMode authenticationMode) {
+      String username,
+      String password,
+      int port,
+      AuthenticationMode authenticationMode,
+      boolean useTLS) {
     this.username = username;
     this.password = password;
-    container = new FixedHostPortGenericContainer("bitnami/memcached:1.6.21");
+    this.useTLS = useTLS;
+
+    if (useTLS) {
+      if (authenticationMode != AuthenticationMode.NONE) {
+        throw new RuntimeException(
+            "Authentication is not currently supported with the TLS-enabled container");
+      }
+
+      container = setupTLSContainer();
+    } else {
+      container = setupContainer(username, password, authenticationMode);
+    }
+
     if (port != DEFAULT_PORT) {
       container.withFixedExposedPort(DEFAULT_PORT, port);
     } else {
       container.withExposedPorts(DEFAULT_PORT);
     }
+
+    start(authenticationMode);
+  }
+
+  private FixedHostPortGenericContainer setupContainer(
+      String username, String password, AuthenticationMode authenticationMode) {
+    final FixedHostPortGenericContainer container =
+        new FixedHostPortGenericContainer("bitnami/memcached" + ":" + MEMCACHED_VERSION);
+
     switch (authenticationMode) {
-      case NONE:
```

Comment:
Could this be NONE when not using TLS?

Reply:
It could, but there is no way to make this check exhaustive (fail in compile time) and for NONE we don't modify the state, so it didn't seem useful to have it. I'll return it though if you prefer to have it.

Reply:
@protocol7 any concerns here? Something else preventing thus PR to be merged? 
Also can we run a CI build to make sure it's all good